frontend/js/pages/BattleDetail.js

```diff
@@ -13,34 +13,46 @@ function BattleDetail(props) {
   const { id } = useParams();
   const { battle } = props.battle;
   const { user } = props.user;
-
+  const fetchedBattle = props.location.query;
   useEffect(() => {
     if (!user) {
       props.getCurrentUser();
     }
-    props.getBattle(id);
+    if (!fetchedBattle) {
+      props.getBattle(id);
+    }
```

Comment:
Put this another `useEffect`, the intention here is to separate concerns into different hooks: https://pt-br.reactjs.org/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns

Reply:
On the line of my previous comment, this should be removed

Reply:
Thanks!